### DISCUSSION TITLE
[Question][Hw2] Issues when retrieving records via mongo shell

### DISCUSSION THREAD

**Discussion by gunsodo** [prosyslab-classroom/cs348-information-security#499] [Q&A]
> **TL;DR** Check your typo in the `mongo` command. Read P.S.!!!

I have written some test code to check whether my attack is running. I decided to check it via `mongo` shell. Apparently, I am stuck with not knowing what exactly is going on.

I installed `mongodb-org` by following [this link](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-ubuntu/#uninstall-mongodb-community-edition) and what I have now in my `Attacker` machine is `MongoDB shell version v4.4.5`. And when I tried to execute the command `mongo -u cs448 -p cs448 102.168.20.1:27017/amp`, these warnings popped up.

```
WARNING: No implicit session: Logical Sessions are only supported on server versions 3.6 and greater.
Implicit session: dummy session
MongoDB server version: 2.6.10
WARNING: shell and server versions do not match
```
When I typed in `db.attack_session.find({})`, it outputs nothing. So now I have no idea what is going on: my code did not work or it is due to the dummy session.

I researched and understood from [this link](https://dba.stackexchange.com/questions/246289/show-collections-in-mongodb-throwing) that it is the version issue. So question number 1, **should I re-install `mongodb` but now with v2.6.10?** And if that is the solution, as far as I know, `mongoDB` does not support v2.6.10 anymore and to install it might be slightly more complicated, question number 2: **does anyone have proper installing instruction for that?**

Question number 3, **is there any workaround for this problem? For example, installing mongo shell with alternative ways?** Maybe I am not on the correct path.

I profoundly apologize for my newbie question. Any kind of help is highly appreciated.

P.S. I think there is typo in the assignment instruction.
```
mongo -u cs448-p cs448 192.168.20.1:27017/amp
```
should be
```
mongo -u cs448 -p cs448 192.168.20.1:27017/amp
```

**Comment by mrpositron** [comment]
Oooh. I had an error due to the typo in the assignment. Thanks that you pointed out for a typo! By the way, installing mongodb-clients is enough.

**Comment by gunsodo** [comment]
> Oooh. I had the similar error due to the typo in the assignment. Thanks that you pointed out for a typo! By the way, installing mongodb-clients is enough.

@MrPositron  I actually tried and it had an error like [this](https://stackoverflow.com/questions/26211671/failed-to-connect-to-127-0-0-127017-reason-errno111-connection-refused). Maybe I uninstall `mongodb-org` and will try again soon.

**Comment by mrpositron** [comment]
Did you do it after fixing a typo? Because I also observed same error message, but after fixing the command it worked OK.

**Comment by gunsodo** [comment]
Oh it's working now. Thank you very much, I just wasted so much time.
